src/plugins/intel_gpu/src/graph/graph_optimizer/compile_graph.cpp

```diff
@@ -36,6 +37,11 @@ void compile_graph::run(program& p) {
         bool can_select_impl = !node->is_type<data>() &&
                                !(node->is_type<mutable_data>() && node->get_dependencies().empty()) &&
                                (!node->is_dynamic() || node->type()->does_dynamic_implementation_exist(*node));
+
+        // TODO: Remove this WA once we have shape agnostic reshape kernel
+        if (node->is_type<reshape>() && node->is_dynamic() && !node->can_be_optimized())
```

Comment:
Dont we need to optimize out reahape in handle_reshape?

Reply:
Probably no. handle reshape is done on early graph optimization stages, so we can't be sure that optimized flag won't be invalidated after fusions or some other passes.